### ISSUE TITLE
Are there any drawbacks to tightly coupling the json-schema to the view? 

### ISSUE THREAD

**Issue by mkmoisen** [ulion/jsonform#18]
I wrote a detailed [question on stackoverflow](http://stackoverflow.com/questions/31777279/adding-properties-to-fields-in-a-json-schema-to-tightly-couple-the-schema-with-a) if anyone could answer, but I'll ask the concise version here:

I would love to leverage something like jsonform so that my UI team could generate a UI form based off the json-schema I provide them. This would cut down tremendously on the hours spent in meetings describing business requirements or how the UI needs to flow. However, this appears to me to tightly couples the backend to the UI. Is there any drawback to this approach or concerns I should look out for?

Best regards!

(I am cross posting this in the 3 jsonform's I see on github for some more visibility)

**Comment by ulion** [comment]
no, you are not, you should always have to use the form definition of
jsonform to adjust the ui to layout as what it want it to be, jsonform only
rely on the schema to generate the form if you did not supply form
definition.

2015-08-03 6:40 GMT+08:00 mkmoisen notifications@github.com:

> I wrote a detailed question on stackoverflow
> http://stackoverflow.com/questions/31777279/adding-properties-to-fields-in-a-json-schema-to-tightly-couple-the-schema-with-a
> if anyone could answer, but I'll ask the concise version here:
> 
> I would love to leverage something like jsonform so that my UI team could
> generate a UI form based off the json-schema I provide them. This would cut
> down tremendously on the hours spent in meetings describing business
> requirements or how the UI needs to flow. However, this appears to me to
> tightly couples the backend to the UI. Is there any drawback to this
> approach or concerns I should look out for?
> 
> Best regards!
> 
> (I am cross posting this in the 3 jsonform's I see on github for some more
> visibility)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ulion/jsonform/issues/18.

## 

Ulion
